apps/compartmentConverter.cpp

```diff
@@ -162,9 +163,22 @@ int main( const int argc, char** argv )
     const brion::CompartmentCounts& counts = in.getCompartmentCounts();
     const brion::GIDSet& gids = in.getGIDs();
 
-    const lunchbox::URI outURI( vm.count( "output" ) == 1 ?
-                                vm[ "output" ].as< std::string >() :
-                                std::string( "out.h5" ));
+    lunchbox::URI outURI( vm.count( "output" ) == 1 ?
+                          vm[ "output" ].as< std::string >() :
+                          std::string( "out.h5" ));
+    if( outURI.getPath().empty( ))
+    {
+        try
+        {
+            outURI.setPath( boost::filesystem::canonical(
+                                inURI.getPath( )).generic_string( ));
+        }
+        catch( const boost::filesystem::filesystem_error& )
+        {
+            // For non-filebased reports, the canonical above will throw.
+            outURI.setPath( inURI.getPath( ));
```

Comment:
I don't think I understand what this path setting is about. Is it trying to place the output file in the same folder as the input file? But then if the inURI is not a file, and the above throws, what is the point of the second line? Won't it be empty anyway?